x-pack/test/security_solution_cypress/cypress/e2e/entity_analytics/asset_criticality/host_flyout.cy.ts

```diff
@@ -0,0 +1,105 @@
+  HOST_DETAILS_FLYOUT_ASSET_CRITICALITY_BUTTON,
+  HOST_DETAILS_FLYOUT_ASSET_CRITICALITY_MODAL_TITLE,
+  HOST_DETAILS_FLYOUT_ASSET_CRITICALITY_MODAL_SELECT,
+  HOST_DETAILS_FLYOUT_ASSET_CRITICALITY_MODAL_SELECT_OPTION,
+  HOST_DETAILS_FLYOUT_ASSET_CRITICALITY_MODAL_SAVE_BTN,
+  HOST_DETAILS_FLYOUT_ASSET_CRITICALITY_LEVEL,
+} from '../../../screens/asset_criticality/flyouts';
+import { deleteAlertsAndRules } from '../../../tasks/api_calls/common';
+import {
+  expandFirstAlertHostFlyout,
+  toggleAssetCriticalityAccordion,
+  toggleAssetCriticalityModal,
+} from '../../../tasks/asset_criticality/common';
+import { createRule } from '../../../tasks/api_calls/rules';
+import { waitForAlertsToPopulate } from '../../../tasks/create_new_rule';
+import { login } from '../../../tasks/login';
+import { visit } from '../../../tasks/navigation';
+import { ALERTS_URL } from '../../../urls/navigation';
+
+describe(
+  'Host details flyout',
+  {
+    tags: ['@ess', '@serverless'],
+    env: {
+      ftrConfig: {
+        kbnServerArgs: [
+          `--xpack.securitySolution.enableExperimental=${JSON.stringify([
+            'entityAnalyticsAssetCriticalityEnabled',
+          ])}`,
+        ],
+      },
+    },
+  },
+  () => {
+    const rule = { ...getNewRule(), investigation_fields: { field_names: ['host.os.name'] } };
+
+    beforeEach(() => {
+      deleteAlertsAndRules();
+      login();
+      createRule(rule);
+      visit(ALERTS_URL);
+      waitForAlertsToPopulate();
+    });
+
+    describe('Host flyout', () => {
+      beforeEach(() => {
+        expandFirstAlertHostFlyout();
+      });
+      it('should display header section', () => {
+        cy.log('header and content');
+
+        cy.get(HOST_DETAILS_FLYOUT_SECTION_HEADER).should('contain.text', 'Host details');
+      });
+
+      it('should display asset criticality accordion', () => {
+        cy.log('asset criticality');
+
+        cy.get(HOST_DETAILS_FLYOUT_ASSET_CRITICALITY_SELECTOR).should(
+          'contain.text',
+          'Asset Criticality'
+        );
+
+        toggleAssetCriticalityAccordion();
+        cy.get(HOST_DETAILS_FLYOUT_ASSET_CRITICALITY_BUTTON).should('have.text', 'Create');
+      });
+
+      it('should display asset criticality modal', () => {
+        cy.log('asset criticality modal');
+
+        toggleAssetCriticalityModal();
+        cy.get(HOST_DETAILS_FLYOUT_ASSET_CRITICALITY_MODAL_TITLE).should(
+          'have.text',
+          'Pick asset criticality level'
+        );
+      });
+
+      it('should update asset criticality state', () => {
+        cy.log('asset criticality update');
+
+        toggleAssetCriticalityModal();
+        cy.get(HOST_DETAILS_FLYOUT_ASSET_CRITICALITY_MODAL_SELECT).should('be.visible').click();
+
+        cy.get(HOST_DETAILS_FLYOUT_ASSET_CRITICALITY_MODAL_SELECT_OPTION)
+          .contains('Important')
+          .click();
+
+        cy.get(HOST_DETAILS_FLYOUT_ASSET_CRITICALITY_MODAL_SAVE_BTN).should('be.visible').click();
```

Comment:
NIT: Wrap it into a meaningful task method. You don't need to check for the element to be visible, cypress makes sure the element is actionable before performing the click.